generator_process.py

```diff
@@ -0,0 +1,93 @@
+import json
+import subprocess
+import sys
+import os
+import numpy as np
+
+class GeneratorProcess():
+    def __init__(self):
+        self.process = subprocess.Popen([sys.executable,'generator_process.py'],cwd=os.path.dirname(os.path.realpath(__file__)),stdin=subprocess.PIPE,stdout=subprocess.PIPE)
+    
+    def kill(self):
+        self.process.kill()
+    
+    def prompt2image(self, args, step_callback, image_callback):
+        b = bytes(json.dumps(args), "utf-8")
+        self.process.stdin.write(len(b).to_bytes(8,sys.byteorder,signed=False))
+        self.process.stdin.write(b)
+        self.process.stdin.flush()
+        if image_callback:
+            for i in range(0,args['iterations']):
+                image_callback(*self.get_image())
+
+    def get_image(self):
+        seed = int.from_bytes(self.process.stdout.read(4),sys.byteorder,signed=False)
+        width = int.from_bytes(self.process.stdout.read(4),sys.byteorder,signed=False)
+        height = int.from_bytes(self.process.stdout.read(4),sys.byteorder,signed=False)
+        image = np.frombuffer(self.process.stdout.read(width*height*4*4),dtype=np.float32)
+        return (seed, width, height, image)
+
+
+
+def main():
+    from absolute_path import absolute_path
+    from stable_diffusion.ldm.generate import Generate
+    from omegaconf import OmegaConf
+    from PIL import ImageOps
+
+    # Support Apple Silicon GPUs as much as possible.
+    os.environ["PYTORCH_ENABLE_MPS_FALLBACK"] = "1"
```

Comment:
This needs to be moved above the generate import to work for me:
```suggestion
    # Support Apple Silicon GPUs as much as possible.
    os.environ["PYTORCH_ENABLE_MPS_FALLBACK"] = "1"

    from stable_diffusion.ldm.generate import Generate
    from omegaconf import OmegaConf
    from PIL import ImageOps
```